src/linear_algebra/matrix/orthogonal.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import data.matrix.basic
+
+/-!
+# Orthogonal
+
+This file contains definitions and properties concerning orthogonality of rows and columns.
+
+## Main results
+
+- `matrix.has_orthogonal_rows`:
+  `A.has_orthogonal_rows` means `A` has orthogonal (with respect to `dot_product`) rows.
+- `matrix.has_orthogonal_cols`:
+  `A.has_orthogonal_cols` means `A` has orthogonal (with respect to `dot_product`) columns.
+
+## Tags
+
+orthogonal
+-/
+
+namespace matrix
+
+variables {α n m : Type*}
+variables [has_mul α] [add_comm_monoid α]
+variables (A : matrix m n α)
+
+open_locale matrix
+
+/-- `A.has_orthogonal_rows` means matrix `A` has orthogonal
+    (with respect to `dot_product`) rows. -/
+def has_orthogonal_rows [fintype n] : Prop :=
+∀ ⦃i₁ i₂⦄, i₁ ≠ i₂ → dot_product (A i₁) (A i₂) = 0
+
+/-- `A.has_orthogonal_cols` means matrix `A` has orthogonal
+    (with respect to `dot_product`) columns. -/
```

Comment:
`dot_product` needs to be prefixed with `matrix.`, this word order reads better:
```suggestion
/-- `A.has_orthogonal_rows` means matrix `A` has orthogonal rows (with respect to
`matrix.dot_product`). -/
def has_orthogonal_rows [fintype n] : Prop :=
∀ ⦃i₁ i₂⦄, i₁ ≠ i₂ → dot_product (A i₁) (A i₂) = 0

/-- `A.has_orthogonal_rows` means matrix `A` has orthogonal columns (with respect to
`matrix.dot_product`). -/
```